templates/statefulset.yaml

```diff
@@ -71,12 +71,14 @@ spec:
             {{- end}}
           - name: DEST_TRANSFORM_URL
             value: "http://{{ include "transformer.fullname" . }}:{{ .Values.transformer.service.port}}"
+          {{- if .Values.postgresql.enabled }}
           - name: COMPUTE_DB_HOST_IN_K8S
             value: "true"
           - name: POSTGRES_POD_NAME
-            value: "{{ .Release.Name }}-{{ .Values.postgresql.nameOverride }}"
+            value: "{{ include "postgresql.host.main" . }}"
           - name: POSTGRES_HEADLESS_SVC
-            value: "{{ .Release.Name }}-{{ .Values.postgresql.nameOverride }}-headless"
+            value: "{{ include "postgresql.host.headless" . }}"
+          {{- end }}
           - name: JOBS_DB_HOST
             value: "This is expected to be set by docker entrypoint script"
```

Comment:
`JOBS_DB_HOST` env is how the server connects to the postgres db. This env value is overwritten by the docker entrypoint script in server if `COMPUTE_DB_HOST_IN_K8S`.
When postgres.enabled is false, we should have this env pointing to the external db's host that is used.
We can do this by having an extra config value like `values.backend.exteranlDBHost`.